pyhive/hive.py

```diff
@@ -116,7 +124,16 @@ def sasl_factory():
                     raise AssertionError
                 sasl_client.init()
                 return sasl_client
-            self._transport = thrift_sasl.TSaslClientTransport(sasl_factory, sasl_auth, socket)
+		
+            if mode == "HTTP" and auth == "NONE":
+                uri = "http://{}:{}/{}".format(host, port, http_path)
+                self._transport = thrift.transport.THttpClient.THttpClient(uri)
+                ap = "%s:%s" % (username,password)
```

Comment:
What do ap and cr stand for? I can't figure out if they're standard abbreviations or not.